src/sql/workbench/electron-browser/modelComponents/table.component.ts

```diff
@@ -159,9 +160,43 @@ export default class TableComponent extends ComponentBase implements IComponent,
 	private layoutTable(): void {
 		let width: number = this.convertSizeToNumber(this.width);
 		let height: number = this.convertSizeToNumber(this.height);
+		let forceFit: boolean = true;
+
+		// convert the tri-state viewmodel columnSizingMode to be either true or false for SlickGrid
+		switch (this.forceFitColumns) {
+			case ColumnSizingMode.Default: {
+				forceFit = false;
+				break;
+			}
+			case ColumnSizingMode.ForceFit: {
+				forceFit = true;
+				break;
+			}
+			case ColumnSizingMode.AutoFit: {
+				// determine if force fit should be on or off based on the number of columns
+				// this can be made more sophisticated if need be in the future.  a simple
+				// check for 3 or less force fits causes the small number of columns to fill the
+				// screen better.  4 or more, slickgrid seems to do a good job filling the view and having forceFit
+				// false enables the scroll bar and avoids the over-packing should there be a very large
+				// number of columns
+				if (this._table.columns.length > 3) {
+					forceFit = false;
+				}
+				else {
+					forceFit = true;
+				}
+				break;
+			}
```

Comment:
Should typically have a default. Normally I'd suggest throwing an error if it's an unrecognized value but in this case it might be fine to just log an error and then set it to Default